service/jmx/src/main/java/org/jolokia/service/jmx/handler/ExecHandler.java

```diff
@@ -73,12 +73,20 @@ public Object doHandleSingleServerRequest(MBeanServerConnection server, JolokiaE
         Object[] params = new Object[nrParams];
         List<Object> args = request.getArguments();
         verifyArguments(request, types, nrParams, args);
-        for (int i = 0;i < nrParams; i++) {
-        	if (types.paramOpenTypes != null && types.paramOpenTypes[i] != null) {
-        		params[i] = context.getMandatoryService(Serializer.class).deserializeOpenType(types.paramOpenTypes[i], args.get(i));
-        	} else { 
-        		params[i] = context.getMandatoryService(Serializer.class).deserialize(types.paramClasses[i], args.get(i));
-        	}
+        List<String> clientTypes = splitOperation(request.getOperation());
+        for (int i = 0; i < nrParams; i++) {
+            if (types.paramOpenTypes != null && types.paramOpenTypes[i] != null) {
+                params[i] = context.getMandatoryService(Serializer.class).deserializeOpenType(types.paramOpenTypes[i], args.get(i));
+            } else {
+                String type;
+                String clientType = clientTypes.size() > i + 1 ? clientTypes.get(i + 1) : null;
+                if (clientType != null && !clientType.equals(types.paramClasses[i])) {
```

Comment:
I added a test case below instead of example, i guess it s better. Let me know if something is not clear in there. 

About you suggestion, yup it looks better than my initial proposal will push another change for that.